buildSrc/src/main/java/org/apache/kafka/message/JsonUtil.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.message;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public final class JsonUtil {
+    /**
+     * The Jackson serializer we use for JSON objects.
+     */
+    static final ObjectMapper JSON_SERDE;
+
+    static {
+        JSON_SERDE = new ObjectMapper();
+        JSON_SERDE.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        JSON_SERDE.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        JSON_SERDE.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+    }
+
+    static <T> T readJsonFileWithComments(Path path, Class<T> clazz) throws IOException {
+        StringBuilder bld = new StringBuilder();
+        for (String line : Files.readAllLines(path)) {
+            bld.append(stripComments(line));
+        }
+        return JSON_SERDE.readValue(bld.toString(), clazz);
+    }
+
+    static String stripComments(String input) {
```

Comment:
Should we have unit tests testing this functionality? Looks like a tricky function

Reply:
It turns out that Jackson has a built-in way to strip comments, so I just enabled this feature.

I added some unit tests for some of the other functions here.